pysqlexpr/table.py

```diff
@@ -183,22 +183,22 @@ class Column:
     __slots__ = ("name", "data_type", "nullable", "default", "description")
 
     name: Identifier
-    data_type: DataType
+    data_type: DataType | type
```

Comment:
Can you help elaborate what arguments do we expect to pass that are instances of `type`? Today, columns are expected to have a `DataType`, which corresponds to typical Snowflake types like boolean, number, string or variant (all of which are defined as sub-classes of `DataType`). You cannot pass Python types.

Reply:
I'm trying to use [strong_typing.inspection's get_class_properties](https://github.com/hunyadi/strong_typing/blob/master/strong_typing/inspection.py#L568)(typ: type) -> Iterable[Tuple[str, type]]

This returns an iterable with column name (`str`) & and coumn type with `type`.
This can be passed to pysqlexpr.table.

## Example
```
            for prop_name, prop_type in get_class_properties(field_type):
                column = Column(
                    prop_name, prop_type, nullable=is_type_optional(prop_type)
                )
```

mypy check fails at the moment:  error: Argument 2 to "Column" has incompatible type "type"; expected "DataType"  [arg-type]

Reply:
FYI, type(prop_type) returns <class 'typing._UnionGenericAlias'>

Reply:
You need to pass an instance of one of the `DataType` sub-classes, e.g. `BooleanType`, `NumberType` or `DateType`. These translate to Snowflake SQL types. You can't pass an arbitrary Python type (such as those returned by type inspection of a data-class) because `pysqlexpr` would not know how to map those to SQL types, and unable to generate a `CREATE` statement. This means that your code might need a mapping, e.g. translate a data-class field type into the appropriate `DataType` sub-class instance.